pkg/app/piped/cmd/piped/piped.go

```diff
@@ -170,6 +170,27 @@ func (p *piped) run(ctx context.Context, input cli.Input) (runErr error) {
 		}
 	}
 
+	// Login to OCI registry.
+	if regs := cfg.ChartRegistries; len(regs) > 0 {
+		reg := toolregistry.DefaultRegistry()
+		helm, _, err := reg.Helm(ctx, "")
+		if err != nil {
+			return fmt.Errorf("failed to find helm to login to OCI registry (%w)", err)
+		}
+
+		for _, r := range regs {
+			switch r.Type {
+			case config.OCIHelmChartRegistry:
+				if r.Username != "" && r.Password != "" {
+					err := loginToOCIRegistry(ctx, helm, r.Address, r.Username, r.Password)
+					if err != nil {
+						return err
+					}
+				}
```

Comment:
```suggestion
				if err := loginToOCIRegistry(ctx, helm, r.Address, r.Username, r.Password); err != nil {
					input.Logger.Error(fmt.Fprintf("failed to login to %s chart registry", r.Address), zap.Error(err))
					return err
				}
```
and move r.Username && r.Password check to loginToOCIRegistry function internal

Reply:
> and move r.Username && r.Password check to loginToOCIRegistry function internal

If `r.Username` or `r.Password` is blank, what value should it return in `loginToOCIRegistry` function?

Reply:
If it returns `nil` then `successfully logged in ...` log will be shown but login will not be executed actually.

If it returns `error` then `failed to login ...` log will be shown but it's not error in the specification because  username and password are optional.

Reply:
what I want is to avoid nested if block. So we can have
```go
    if r.Username == "" || r.Password == "" {
        continue
    }
    if err := loginToOCIRegistry(ctx, helm, r.Address, r.Username, r.Password); err != nil { ... }
```
or we can just pass whatever username and password (possible to be blank) to loginToOCIRegistry, and inside that function, just skip login and return nil as you mentioned. wdyt?

Reply:
I thought the former proposal was more concise, so I modified it.

```go
switch r.Type {
case config.OCIHelmChartRegistry:
	if r.Username != "" && r.Password != "" {
		continue
	}

	if err := loginToOCIRegistry(ctx, helm, r.Address, r.Username, r.Password); err != nil {
		input.Logger.Error(fmt.Sprintf("failed to login to %s chart registry", r.Address), zap.Error(err))
		return err
	}
	input.Logger.Info("successfully logged in to Helm chart registry", zap.String("address", r.Address))
}
```